### ISSUE TITLE
Login improvement (FE)

### ISSUE THREAD

**Issue by MariusJulius** [keeleinstituut/tv-tolkevarav#345]

**Comment by KaarelKa** [comment]
Will add loader here for now. Currently we rely only on keycloak to tell us whether the user is logged in or not + We need to do extra request to fetch institutions + select 1 of them, if needed. This results in the delay we see.

@thenouan I could store the same info locally to optimistically show the correct screen right away, but do you see any potential issues with that ? Solution would be sth like:

1. Store the refresh token, refresh token expiration time and selected institution id locally.
2. Check if refresh token is expired or not
3. If yes, then consider the user logged out
4. If no, then do the institution select query, with the stored refresh token and stored institution id
5. Set up ApiClient with the new auth token (Assuming refresh token endpoint returns the new auth token directly, haven't checked yet)
6. Initialize keycloak

Basically this would reduce the loading time quite significantly, but I'm not 100% sure if there might be any issues with this, since keycloak would not be initialized on FE while doing this.

**Comment by plakitkelly** [comment]
I remember seeing the main page for a few seconds before and now it's much better. But I still can see main page for a moment. Is it possible to remove this or is this the best that can be done? @KaarelKa 

https://github.com/keeleinstituut/tv-tolkevarav/assets/130659408/04288f88-2414-4b89-820c-2bb09ed785fb

**Comment by plakitkelly** [comment]
I move this done. Maybe in 2nd part it will be better..
Currently I see landing page on login and logout for a moment.
